### ISSUE TITLE
When calculating experiences, avoid some unnecessary dict accesses

### ISSUE THREAD

**Issue by marco-c** [mozilla/bugbug#493]
We can introduce some temporary variables in some cases to avoid accessing the same elements multiple times.
E.g. https://github.com/mozilla/bugbug/pull/488#discussion_r287302102.

**Comment by bilalib** [comment]
I'd like to work on this. Does this require new test cases? And could you point me to a line where this kind of optimization is possible?

**Comment by marco-c** [comment]
> Does this require new test cases?

No, we already have some covering the code you will have to change.

> And could you point me to a line where this kind of optimization is possible?

There is a link in the first comment :)

**Comment by hmrishavbandy** [comment]
@marco-c Hey i would like to work on this. I am completely new to contributing anything, so could you please help me out with initial steps? Thanks a lot .

**Comment by marco-c** [comment]
Hello @hmrishavbandy, if you don't know how git/github work at all, I'd suggest reading some online guide/tutorial to start. I think GitHub itself offers some kind of tutorial.

**Comment by hmrishavbandy** [comment]
Yep I have followed github tutorials and formed my own repo (only Readme tho). Can you assign this issue to me?

**Comment by marco-c** [comment]
Feel free to work on the issue. Any issue which has no PR linked to it is considered as unassigned (see https://github.com/mozilla/bugbug/blob/master/CONTRIBUTING.md).
